### ISSUE TITLE
tests: kernel.workqueue.api tests fail on multiple platforms

### ISSUE THREAD

**Issue by PerMac** [zephyrproject-rtos/zephyr#29329]
**Describe the bug**
Tests "work_submit_to_multipleq" and "work_resubmit_nowait_to_queue " from "tests/kernel/workq/work_queue_api" fail on nrf platforms (nrf52840dk_nrf52840, nrf52dk_nrf52832 and nrf9160dk_nrf9160)

**To Reproduce**
1. go to zephyr dir
2. call `./scripts/sanitycheck --hardware-map map.yml --device-testing -T tests/kernel/workq/work_queue_api/`
3. See error

**Expected behavior**
The tests pass

**Impact**
Not clear

**Logs and console output**

```
 START - test_work_submit_to_multipleq Assertion failed at CMAKE_SOURCE_DIR/src/main.c:277: twork_submit_multipleq: k_delayed_work_pending(&new_work) is false FAIL - test_work_submit_to_multipleq
 ```
and
```
START - test_work_resubmit_nowait_to_queue Assertion failed at CMAKE_SOURCE_DIR/src/main.c:357: twork_resubmit_nowait: k_work_pending(&delayed_work_sleepy.work) is true FAIL - test_work_resubmit_nowait_to_queue 
```

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - Toolchain Zephyr SDK v0.11.3
 - Zephyr version: zephyr-v2.4.0-685-g7e64fba9ed

**Comment by maksimmasalski** [comment]
#29328 Same thing

**Comment by nashif** [comment]
bisection leads to bd76994dc21b3b6a7a2d6a02413c20cab82762ce, FYI @pabigot

**Comment by nashif** [comment]
does not seem to be nrf specific.

**Comment by pabigot** [comment]
I've extracted the fixes for these that were in #28891 since that's stalled.
